src/herbivoresAndCarnivores.js

```diff
@@ -2,14 +2,39 @@
 
 class Animal {
   // write your code here
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
 }
 
+Animal.alive = [];
+
 class Herbivore extends Animal {
   // write your code here
+  constructor() {
+    super();
+    this.hidden = false;
+  }
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
   // write your code here
+  bite(herbivore) {
+    if (herbivore instanceof Herbivore && herbivore.hidden === false) {
+      herbivore.health -= 50;
+    }
+
+    if (herbivore.health <= 0) {
+      const indOfDead = Animal.alive.findIndex(animal => animal === herbivore);
+
+      Animal.alive.splice(indOfDead, 1);
+    }
```

Comment:
When you are using indexOf and splice combo - you should check if element was found before deleting element. (first item in the checklist)